### PR TITLE
Added implementation of the Guided IG attribution method.

### PR DESCRIPTION
**Pull request by gkapish**
- added implementation of the Guided IG attribution method.
- removed support for PY2 and versions of PY3 before 3.5.

### CODE REVIEW

File: saliency/core/guided_ig.py [hunk 1]
```diff
@@ -0,0 +1,241 @@
...
+    # `l1_target` is the desired L1 distance after completion of this step.
+    l1_target = l1_total * (1 - (step + 1) / steps)
+
+    # Iterate until the desired L1 distance has been reached.
+    gamma = np.inf
+    while gamma > 1.0:
+      x_old = x.copy()
+      x_alpha = translate_x_to_alpha(x, x_input, x_baseline)
+      x_alpha[np.isnan(x_alpha)] = alpha_max
+      # All features that felt too behind in terms of alpha, should be assigned
```
Comment (bwedin) [inline saliency/core/guided_ig.py:128]: "fell too behind"?

File: setup.py [hunk 2]
```diff
@@ -22,7 +22,7 @@
     # Versions should comply with PEP440.  For a discussion on single-sourcing
     # the version across setup.py and the project code, see
     # https://packaging.python.org/en/latest/single_source_version.html
-    version='0.1.2',
+    version='0.1.3',
```
Comment (bwedin) [inline setup.py:25]: Unsure if the version bump should go in this PR or a separate one, since there may be a couple other smaller PRs as well.

**Review by bwedin — COMMENTED** [review]
I'm seeing a couple issues with the Colab:

1) The headers are mismatched (i.e. XRAI uses "#", Guided IG uses "###"). Can we make these consistent?
2) The TF1 model appears to be missing or not loading correctly, the TFHub folks may have deprecated it. Can you look into it?

File: saliency/core/guided_ig.py (hunk 1 above)
Comment (gkapish) [reply saliency/core/guided_ig.py:128]: Thanks for noticing it.

File: setup.py (hunk 2 above)
Comment (gkapish) [reply setup.py:25]: What is the policy of changing the version? This change looks like a substantial one to justify bumping the version.

Comment (tolga-b) [reply setup.py:25]: We usually bump version in separate follow up PRs, can you first check this in then create a follow up one?

Comment (gkapish) [reply setup.py:25]: Sure.

File: saliency/core/guided_ig_test.py [hunk 3]
```diff
@@ -0,0 +1,162 @@
...
+
+"""Tests the core implementation of Guided IG."""
+import unittest
+
+from . import guided_ig
+from .base import INPUT_OUTPUT_GRADIENTS
+import numpy as np
+
+
+class GuidedIGTest(unittest.TestCase):
```
Comment (bwedin) [inline saliency/core/guided_ig_test.py:23]: Can you add a test which checks for correctness when baseline > input?

**Review by bwedin — COMMENTED** [review]
Tests/structure of code looks good overall, @tolga-b can you check the algorithm itself to make sure it's correct wrt the paper?

File: saliency/core/guided_ig_test.py (hunk 3 above)
Comment (gkapish) [reply saliency/core/guided_ig_test.py:23]: Good suggestion. Done.

**Comment by gkapish** [comment]
All outstanding comments should be addressed now.

**Comment by tolga-b** [comment]
@gkapish Thank you for this! Could you please update the description and list what this PR does (e.g. remove support for PY2.x, Guided IG, major things) . I will follow-up with the review of the algorithm.

**Review by tolga-b — APPROVED** [review]
Did a high level review, approving to unblock.

**Pull request merged**